src/combinations.jl

```diff
@@ -11,8 +11,15 @@ struct Combinations{T}
     t::Int
 end
 
-Base.start(c::Combinations) = collect(1:c.t)
-function Base.next(c::Combinations, s)
+function Base.iterate(c::Combinations)
```

Comment:
Rather than copying the code to a separate method, I believe you could just use `Base.iterate(c::Combinations, s = collect(1:c.t))`. 🙂

Reply:
Thanks!!!! For some reason I don't truly understand this actually fixed the problem!

I'll now carry on with the rest.

Reply:
No problem! The default state (second argument) of the `iterate` function is supposed to be the initial iteration state, which is what `start` previously provided.